qupulse/_program/waveforms.py

```diff
@@ -243,6 +243,43 @@ def __repr__(self):
         return f'{type(self).__name__}(channel={self._channel_id}, waveform_table={self._table})'
 
 
+class ConstantWaveform(Waveform):  
+    def __init__(self, duration: float, amplitude: Any, channel: ChannelID):
+        """ Create a qupulse waveform from a qupulse template that has a QI waveform inside """
+        self._duration = duration
+        self._amplitude = amplitude
+        self._channel = channel
+
+        self._is_constant_waveform = True
+
+    @property
+    def duration(self) -> TimeType:
+        return time_from_float(float(self._duration), absolute_error=PULSE_TO_WAVEFORM_ERROR)
+
+    @property
+    def defined_channels(self) -> Set[ChannelID]:
+        """The channels this waveform should played on. Use
+            :func:`~qupulse.pulses.instructions.get_measurement_windows` to get a waveform for a subset of these."""
+
+        return {self._channel}
+
+    def compare_key(self) -> str:
+        return 'ConstantWaveform:' + cast(str, hash([self._duration, self._amplitude, self._channel]))
```

Comment:
Typically a tuple of hashables is returned here that is independent of the running instance since the result of `hash(obj)` should change with each interpreter restart.